### PR TITLE
Define new version manifest variables

### PR DESCRIPTION
**Pull request by mthalman**
This extends the version variable pattern that already exists for .NET 9 (originally introduced in https://github.com/dotnet/dotnet-docker/pull/5039) to the entire manifest. This gets all repos and versions consistent with their version variable usage.

This allows for the ability to more easily change names in one location. See https://github.com/dotnet/dotnet-docker/pull/5579#discussion_r1646521607.

### CODE REVIEW

**Review by MichaelSimons — APPROVED** [review]

File: README.sdk.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ Tags | Dockerfile | OS Version
 8.0.302-cbl-mariner2.0-amd64, 8.0-cbl-mariner2.0-amd64, 8.0.302-cbl-mariner2.0, 8.0-cbl-mariner2.0 | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/nightly/src/sdk/8.0/cbl-mariner2.0/amd64/Dockerfile) | CBL-Mariner 2.0
 8.0.302-cbl-mariner2.0-aot-amd64, 8.0-cbl-mariner2.0-aot-amd64, 8.0.302-cbl-mariner2.0-aot, 8.0-cbl-mariner2.0-aot | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/nightly/src/sdk/8.0/cbl-mariner2.0-aot/amd64/Dockerfile) | CBL-Mariner 2.0
 6.0.423-bookworm-slim-amd64, 6.0-bookworm-slim-amd64, 6.0.423-bookworm-slim, 6.0-bookworm-slim | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/nightly/src/sdk/6.0/bookworm-slim/amd64/Dockerfile) | Debian 12
-6.0.423-bullseye-slim-amd64, 6.0-bullseye-slim-amd64, 6.0.423-bullseye-slim, 6.0-bullseye-slim, 6.0.423-1, 6.0 | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/nightly/src/sdk/6.0/bullseye-slim/amd64/Dockerfile) | Debian 11
+6.0.423-bullseye-slim-amd64, 6.0-bullseye-slim-amd64, 6.0.423-bullseye-slim, 6.0-bullseye-slim, 6.0.423, 6.0 | [Dockerfile](https://github.com/dotnet/dotnet-docker/blob/nightly/src/sdk/6.0/bullseye-slim/amd64/Dockerfile) | Debian 11
```
Comment (mthalman) [inline README.sdk.md:86]: This ended up fixing a mistake from https://github.com/dotnet/dotnet-docker/pull/5614 that didn't remove the `-1` from the `6.0.423` tag.

**Review by MichaelSimons — APPROVED** [review]

**Review by lbussell — CHANGES_REQUESTED** [review]
Tags in `customBuildLegGroups` need to be updated for .NET 6 and 8. Example from 9.0: https://github.com/dotnet/dotnet-docker/blob/56dad2cfdbc0a0e3d728a34f2934f5ccf5551a0c/manifest.json#L2028

**Review by lbussell — APPROVED** [review]
:shipit:

**Comment by lbussell** [comment]
Needed to re-do the Alpine 3.20 work in order to get a clean merge here. I also caught the `extra`, `aot`, and `composite` variants floating tags were not moved to 3.20. So this includes that change as well.

**Pull request merged**